csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -1653,6 +1657,214 @@ private void TestInferenceSessionWithByteArray()
+                    chunk.Dispose();
+                    alloc_info.Dispose();
+                }
+            }
+        }
+
+#if USE_CUDA
+        void TestCUDAAllocator(InferenceSession session)
+        {
+            int device_id = 0;
+            using (var info_cuda = new OrtMemoryInfo(OrtMemoryInfo.allocatorCUDA, OrtAllocatorType.ArenaAllocator, device_id, OrtMemType.Default))
+            {
+                Assert.NotEqual(info_cuda.Pointer, IntPtr.Zero);
+                Assert.Equal("Cuda", info_cuda.Name);
+                Assert.Equal(device_id, info_cuda.Id);
+                Assert.Equal(OrtAllocatorType.ArenaAllocator, info_cuda.GetAllocatorType());
+                Assert.Equal(OrtMemType.Default, info_cuda.GetMemoryType());
+
+                using (var allocator = new OrtAllocator(session, info_cuda))
+                {
+                    Assert.NotEqual(allocator.Pointer, IntPtr.Zero);
+                    var alloc_info = allocator.Info;
+                    Assert.True(info_cuda.CompareMemoryInfo(alloc_info));
+
+                    uint size = 1024;
+                    OrtMemoryAllocation chunk = allocator.Allocate(size);
+                    Assert.NotEqual(chunk.Pointer, IntPtr.Zero);
+                    Assert.Equal(chunk.Size, size);
+                    Assert.True(chunk.Info.CompareMemoryInfo(alloc_info));
+                    chunk.Dispose();
+                    alloc_info.Dispose();
+                }
+            }
+        }
+#endif
+
+        [Fact]
+        private void TestAllocator()
+        {
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "squeezenet.onnx");
+            using (SessionOptions options = new SessionOptions())
+            {
+                options.AppendExecutionProvider_CPU(1);
+#if USE_CUDA
+                options.AppendExecutionProvider_CUDA(0);
+#endif
+                using (var session = new InferenceSession(modelPath, options))
+                {
+                    TestCPUAllocator(session);
+#if USE_CUDA
+                    TestCUDAAllocator(session);
+#endif
+                }
+            }
+        }
+
+        /// <summary>
+        /// This works only for allocations accessible from host memory
+        /// </summary>
+        /// <param name="buffer"></param>
+        /// <param name="elements"></param>
+        private static void PopulateNativeBufferFloat(OrtMemoryAllocation buffer, float[] elements)
+        {
+            if(buffer.Size < elements.Length * sizeof(float))
+            {
+                Assert.True(false);
+            }
+
+            unsafe
+            {
+                float* p = (float*)buffer.Pointer;
+                for (int i = 0; i < elements.Length; ++i)
+                {
+                    *p++ = elements[i];
+                }
+            }
+        }
+
+        private static void CompareNativeBufferFloat(OrtMemoryAllocation buffer, float[] elements, IEqualityComparer<float> comp)
```

Comment:
Should this be Fact ?

Reply:
Nope

---
In reply to: [463255323](https://github.com/microsoft/onnxruntime/pull/4646#discussion_r463255323) [](ancestors = 463255323)